From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
Date: Thu, 23 Dec 2021 23:59:12 -0500
Subject: [PATCH] Fix Entity Position Desync

If entities were teleported in the first tick it would not be send to the client.

diff --git a/src/main/java/net/minecraft/server/level/ServerEntity.java b/src/main/java/net/minecraft/server/level/ServerEntity.java
index 9a6c67b614944f841813ec2892381c3342bc365c..501c5f9026622e1a753db7ced26781985965e408 100644
--- a/src/main/java/net/minecraft/server/level/ServerEntity.java
+++ b/src/main/java/net/minecraft/server/level/ServerEntity.java
@@ -164,7 +164,7 @@ public class ServerEntity {
                 }
                 // CraftBukkit end
 
-                if (this.tickCount > 0 || this.entity instanceof AbstractArrow) {
+                if (this.tickCount >= 0 || this.entity instanceof AbstractArrow) {
                     // Paper start - remove allocation of Vec3D here
                     long k = ClientboundMoveEntityPacket.entityToPacket(vec3d_dx);
                     long l = ClientboundMoveEntityPacket.entityToPacket(vec3d_dy);
